src/cli.js

```diff
@@ -218,17 +219,33 @@ async function runCmd (argv, stdout, stderr) {
       );
       if (existsSync(outDir))
         rimraf.sync(outDir);
+      mkdirp.sync(outDir);
       run = true;
 
     // fallthrough
     case "build":
       if (args._.length > 2)
         errTooManyArguments("build");
 
+      function hasTypeModule (path) {
+        let root = resolve('/');
+        while ((path = resolve(path, '..')) !== root) {
+          try {
+            return JSON.parse(readFileSync(eval('resolve')(path, 'package.json')).toString()).type === 'module';
+          }
+          catch (e) {
+            if (e.code === 'ENOENT')
+              continue;
+            throw e;
+          }
+        }
+      }
+
       let startTime = Date.now();
       let ps;
       const buildFile = eval("require.resolve")(resolve(args._[1] || "."));
-      const ext = buildFile.endsWith('.cjs') ? '.cjs' : '.js';
+      const esm = buildFile.endsWith('.mjs') || buildFile.endsWith('.js') && hasTypeModule(buildFile);
+      const ext = buildFile.endsWith('.cjs') ? '.cjs' : esm && (buildFile.endsWith('.mjs') || !hasTypeModule(buildFile)) ? '.mjs' : '.js';
```

Comment:
I'm curious why you didn't decide to preserve the extension of the input file?

I would expect this:
```
.cjs => .cjs
.mjs => .mjs
.js => .js
```

Reply:
@style we do keep the extension so that `.cjs` -> `.cjs` and `.mjs` -> `.mjs`. The issue with `.js` is that it will be read by webpack as supporting both CJS and ESM, so isn't enough information on its own, hence the boundary check.